pinot-core/src/test/java/com/linkedin/pinot/core/startree/BaseStarTreeIndexTest.java

```diff
@@ -42,95 +42,64 @@
-import org.testng.annotations.BeforeSuite;
-import org.testng.annotations.Test;
+
 
 /**
- * This test generates a Star-Tree segment with random data, and ensures that
- * aggregation results computed using star-tree index operator are the same as
- * aggregation results computed by scanning raw docs.
+ * Base class containing common functionality for all star-tree integration tests.
  */
-public class TestStarTreeIntegrationTest {
-  private static final Logger LOGGER = LoggerFactory.getLogger(TestStarTreeIntegrationTest.class);
+public class BaseStarTreeIndexTest {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BaseStarTreeIndexTest.class);
 
-  private static final String SEGMENT_NAME = "starTreeSegment";
-  private static final String SEGMENT_DIR_NAME = "/tmp/star-tree-index";
   private static final String TIME_COLUMN_NAME = "daysSinceEpoch";
   private static final int NUM_DIMENSIONS = 4;
   private static final int NUM_METRICS = 2;
   private static final int METRIC_MAX_VALUE = 10000;
-  private final long _randomSeed = System.nanoTime();
-
-  private IndexSegment _segment;
-  private Schema _schema;
-
-  @BeforeSuite
-  void setup()
-      throws Exception {
-    buildSegment(SEGMENT_DIR_NAME, SEGMENT_NAME);
-    loadSegment(SEGMENT_DIR_NAME, SEGMENT_NAME);
-  }
-
-  @AfterSuite
-  void tearDown()
-      throws IOException {
-    FileUtils.deleteDirectory(new File(SEGMENT_DIR_NAME));
-  }
-
-  /**
-   * This test ensures that the aggregation result computed using the star-tree index operator
-   * is the same as computed by scanning raw-docs, for a hard-coded set of queries.
-   *
-   * @throws Exception
-   */
-  @Test
-  public void testHardCodedQueries() throws Exception {
-    String[] queries = new String[]{
-        "select sum(m1) from T",
-        "select sum(m1) from T where d1 = 'd1-v1'",
-        "select sum(m1) from T where d1 <> 'd1-v1'",
-        "select sum(m1) from T where d1 between 'd1-v1' and 'd1-v3'",
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2')",
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1')",
-        "select sum(m1) from T group by d1",
-        "select sum(m1) from T group by d1, d2",
-        "select sum(m1) from T where d1 = 'd1-v2' group by d1",
-        "select sum(m1) from T where d1 between 'd1-v1' and 'd1-v3' group by d2",
-        "select sum(m1) from T where d1 = 'd1-v2' group by d2, d3",
-        "select sum(m1) from T where d1 <> 'd1-v1' group by d2",
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') group by d2",
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3",
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3, d4"
-    };
-
+  protected final long _randomSeed = System.nanoTime();
+
+
+  protected String[] _hardCodedQueries =
+      new String[]{
+          "select sum(m1) from T",
```

Comment:
ATM, the test simply aggregates on all metric columns, without looking at the actual aggregations in the query. This was done to be more exhaustive and perform aggregations on all metrics, without needing to explicitly write-down those queries. Will handle count(*) in upcoming PR's.